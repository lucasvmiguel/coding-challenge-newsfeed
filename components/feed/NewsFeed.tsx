import styled from 'styled-components'

import FeedItem, { Item } from './FeedItem'

type Props = {
  items: Item[];
}

export default function NewsFeed({ items }: Props) {
  return (
    <div>
      <ul>
        {items.map(item => (
          <ListItem key={`${item.id}-${item.__typename}`}>
            <FeedItem item={item} />
          </ListItem>
        ))}
      </ul>
    </div>
  )
}

const ListItem = styled.li`
  list-style-type: none;
  padding-bottom: 20px;
`
