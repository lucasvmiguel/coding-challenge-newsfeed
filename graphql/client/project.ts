import { gql } from '@apollo/client'

export const PROJECT_QUERY = gql`
  query project($id: Int!) {
    project(id: $id) {
      id
      name
      description
      icon_url
      users {
        id
        name
        avatar_url
      }
    }
  }
`
