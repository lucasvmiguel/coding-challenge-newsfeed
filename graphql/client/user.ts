import { gql } from '@apollo/client'

export const USER_QUERY = gql`
  query user($id: Int!) {
    user(id: $id) {
      id
      name
      bio
      fellowship
      avatar_url
      projects {
        id
        name
        icon_url
      }
    }
  }
`