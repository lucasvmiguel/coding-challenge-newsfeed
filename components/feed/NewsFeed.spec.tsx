import renderer from 'react-test-renderer'

import { project } from '../../test/mocks/project'
import { user } from '../../test/mocks/user'
import { announcement } from '../../test/mocks/announcement'

import NewsFeed from './NewsFeed'

it('renders NewsFeed component correctly - snapshot', () => {
  const tree = renderer
    .create(<NewsFeed items={[announcement, user, project]} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
