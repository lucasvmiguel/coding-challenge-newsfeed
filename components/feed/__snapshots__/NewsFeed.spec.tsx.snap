// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NewsFeed component correctly - snapshot 1`] = `
<div>
  <ul>
    <li
      className="sc-dQoVA eBdjko"
    >
      <div
        className="sc-bdfBQB eDCjUP"
      >
        <div
          className="sc-dlfnuX iOZAXF"
        >
          <div
            className="sc-hKgJUU jkblDv"
          >
            <img
              className="sc-gsTEea iRQMQw"
              src="https://avatars.dicebear.com/api/initials/a.svg"
            />
          </div>
          <div
            className="sc-eCstlR evEecB"
          >
            <h2
              data-test="announcement-card-title"
            >
              title testing
            </h2>
            <div
              className="sc-jSgvzq"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>body testing</p>
",
                }
              }
            />
          </div>
        </div>
      </div>
    </li>
    <li
      className="sc-dQoVA eBdjko"
    >
      <div
        className="sc-bdfBQB eDCjUP"
      >
        <div
          className="sc-dlfnuX iOZAXF"
        >
          <div
            className="sc-hKgJUU jkblDv"
          >
            <img
              className="sc-gKseQn hvAGSQ"
              src="icon.svg"
            />
          </div>
          <div
            className="sc-eCstlR evEecB"
          >
            <h2
              data-test="user-card-name"
            >
              name testing
            </h2>
            <p>
              Fellowship: 
              founders
            </p>
            <div
              className="sc-jSgvzq"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>desc testing</p>
",
                }
              }
            />
            <h3>
              Projects:
            </h3>
            <div
              className="sc-iBPTik ddGxJq"
            >
              <div
                className="sc-fubCzh kGMoIv"
              >
                <img
                  className="sc-gsTEea iRQMQw"
                  src="icon2.svg"
                />
              </div>
              <div
                className="sc-pGacB dXoMBx"
              >
                <a
                  href="/projects/2"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  user testing
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      className="sc-dQoVA eBdjko"
    >
      <div
        className="sc-bdfBQB eDCjUP"
      >
        <div
          className="sc-dlfnuX iOZAXF"
        >
          <div
            className="sc-hKgJUU jkblDv"
          >
            <img
              className="sc-gsTEea iRQMQw"
              src="icon.svg"
            />
          </div>
          <div
            className="sc-eCstlR evEecB"
          >
            <h2
              data-test="project-card-name"
            >
              name testing
            </h2>
            <div
              className="sc-jSgvzq"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>desc testing</p>
",
                }
              }
            />
            <h3>
              Participants:
            </h3>
            <div
              className="sc-kEjbQP czxUPI"
            >
              <div
                className="sc-iqHYmW fovqHA"
              >
                <img
                  className="sc-jrAFXE dcmrNt"
                  src="icon2.svg"
                />
              </div>
              <div
                className="sc-crrszt iBTWmQ"
              >
                <a
                  href="/users/2"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  user testing
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
